tmt/steps/discover/__init__.py

```diff
@@ -1,24 +1,33 @@
 import os
-from typing import Optional, Type
+from typing import TYPE_CHECKING, Any, List, Optional, Type, cast
 
 import click
 from fmf.utils import listed
 
 import tmt
+from tmt.steps import StepData
+
+if TYPE_CHECKING:
```

Comment:
what is this for?

Reply:
The `TYPE_CHECKING` variable? From my understanding, it helps with reducing the amount of imports. It is `True` only when doing type checking. When the code is being run in production, the imports in this `if` won't be imported as they are not needed - they're used only for typing.

Looking at the code, I'd say even the `from tmt.steps import StepData` line should be in this `if`.

Reply:
This looks like a good idea, I was asking since this condition is not applied in all the other modules where we have type annotation, so there's a consistency issue.. Do we want to add it to all the other modules? or are we okay that some have the conditional imports and some don't?
@thrix

Reply:
It does not need to be in all modules, so it depends.

@janhavlin can you change `StepData` to `tmt.steps.Data` and remove it, seems it would bring some consistency here, or put it under and change other occurances to rather use `StepData`. I agree we should be consistent if possible.

Reply:
Same seems to be applicable to `Test`. Or if you think it helps readability ... but to me more looks like inconsistency ...

Reply:
Agreed